package.json

```diff
@@ -1,6 +1,6 @@
 {
   "name": "nodebb-plugin-calendar",
-  "version": "1.2.1",
+  "version": "1.3.0",
```

Comment:
socket.io ist necessary if you have not installed it globally, tsc will fail without.

Will revert the version feature bump if you take care of it on actual release.

Reply:
That doesn't sound right. I'll have to try it locally. Do you mind telling me where and how it errors?

Reply:
If you're fixing typescript types unrelated to your other changes, put that in a separate commit please.

Reply:
> That doesn't sound right. I'll have to try it locally. Do you mind telling me where and how it errors?

sure here the initial error:
```
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ rm package-lock.json
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ rm -rf node_modules/
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ npm i

added 322 packages, and audited 323 packages in 26s

94 packages are looking for funding
  run `npm fund` for details

2 vulnerabilities (1 low, 1 moderate)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ npm run build

> nodebb-plugin-calendar@1.2.1 build
> npm run compile && webpack --progress --mode=development


> nodebb-plugin-calendar@1.2.1 compile
> tsc -p src/lib

error TS2688: Cannot find type definition file for 'socket.io'.
  The file is in the program because:
    Entry point of type library 'socket.io' specified in compilerOptions

  src/lib/tsconfig.json:16:7
    16       "socket.io"
             ~~~~~~~~~~~
    File is entry point of type library specified here.


Found 1 error.

```

after installing socket.io:
```
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ npm i --save-dev socket.io

added 16 packages, and audited 339 packages in 2s

94 packages are looking for funding
  run `npm fund` for details

2 vulnerabilities (1 low, 1 moderate)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ npm run build

> nodebb-plugin-calendar@1.2.1 build
> npm run compile && webpack --progress --mode=development


> nodebb-plugin-calendar@1.2.1 compile
> tsc -p src/lib

assets by status 4.34 MiB [cached] 211 assets
assets by path . 505 KiB
  asset src_client_eventModal_ts.d197122aac8f05fa23b8.js 198 KiB [emitted] [immutable]
  asset src_client_clientSideTranslation_ts.6224072f80585c25135c.js 112 KiB [emitted] [immutable]
  asset src_calendar_index_ts.f7959fabd3d2d26ec8bd.js 94.3 KiB [emitted] [immutable]
  asset client.js 70.8 KiB [emitted] [compared for emit] (name: client)
  asset src_client_responses_ts.810de5a9c5374e705b37.js 18.6 KiB [emitted] [immutable]
  asset src_lib_translatorModule_ts.00f68eeab78295506f95.js 10.4 KiB [emitted] [immutable]
runtime modules 21.5 KiB 12 modules
modules by path ./node_modules/ 1.61 MiB
  modules by path ./node_modules/moment/ 673 KiB 137 modules
  modules by path ./node_modules/fullcalendar/dist/ 872 KiB 75 modules
  ./node_modules/eonasdan-bootstrap-datetimepicker/src/js/bootstrap-datetimepicker.js 105 KiB [built] [code generated]
modules by path ./src/ 44 KiB
  modules by path ./src/client/*.ts 28.8 KiB 10 modules
  modules by path ./src/calendar/*.ts 7.39 KiB 4 modules
  modules by path ./src/lib/*.ts 7.74 KiB
    ./src/lib/parse.ts 1.85 KiB [built] [code generated]
    + 3 modules
+ 7 modules

ERROR in /home/frozen_byte/Projects/nodebb-plugin-calendar/src/client/tsconfig.json
/home/frozen_byte/Projects/nodebb-plugin-calendar/src/client/tsconfig.json
[tsl] ERROR
      TS2688: Cannot find type definition file for 'socket.io-client'.
  The file is in the program because:
    Entry point of type library 'socket.io-client' specified in compilerOptions
ts-loader-default_e3b0c44298fc1c14

webpack 5.75.0 compiled with 1 error in 3411 ms
```
after installing socket.io-client (just need the typings but that package is deprecated)
```
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ npm i --save-dev socket.io-client

added 3 packages, and audited 342 packages in 3s

94 packages are looking for funding
  run `npm fund` for details

2 vulnerabilities (1 low, 1 moderate)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
frozen_byte@frozen-pc:~/Projects/nodebb-plugin-calendar$ npm run build

> nodebb-plugin-calendar@1.2.1 build
> npm run compile && webpack --progress --mode=development


> nodebb-plugin-calendar@1.2.1 compile
> tsc -p src/lib

assets by status 4.67 MiB [cached] 215 assets
assets by path . 165 KiB
  asset src_calendar_index_ts.fff961832ac7f5a4cfe9.js 94.3 KiB [emitted] [immutable]
  asset client.js 70.8 KiB [emitted] [compared for emit] (name: client)
runtime modules 21.5 KiB 12 modules
modules by path ./node_modules/ 1.61 MiB
  modules by path ./node_modules/moment/ 673 KiB 137 modules
  modules by path ./node_modules/fullcalendar/dist/ 872 KiB 75 modules
  ./node_modules/eonasdan-bootstrap-datetimepicker/src/js/bootstrap-datetimepicker.js 105 KiB [built] [code generated]
modules by path ./src/ 44 KiB
  modules by path ./src/client/*.ts 28.8 KiB 10 modules
  modules by path ./src/calendar/*.ts 7.39 KiB 4 modules
  modules by path ./src/lib/*.ts 7.74 KiB
    ./src/lib/parse.ts 1.85 KiB [built] [code generated]
    + 3 modules
+ 7 modules
webpack 5.75.0 compiled successfully in 3344 ms
```

Reply:
Can you try removing the socket-related stuff from tsconfig instead? I think it's not actually necessary anymore.

Reply:
I am afraid there are plenty other errors without the socket typings all are like this:
```
node_modules/@types/bootstrap/index.d.ts:151:69 - error TS2304: Cannot find name 'HTMLElement'.

151     placement?: Placement | ((this: TooltipInstance<this>, tooltip: HTMLElement, trigger: Element) => Placement);
```

And I will not investigate further. I added the change to an dedicated commit so if you don't like them they are easily reverted before merge.

Reply:
Ok thanks anyways

Reply:
reverted.